opensearch/Dockerfile

```diff
@@ -0,0 +1,145 @@
+echo y | bin/opensearch-plugin install org.opensearch.plugin:performance-analyzer:${OPENSEARCH_PERFORMANCE_ANALYZER}
+
+# The JDK from jdk-base is used instead.
+rm -r jdk
+
+# Change the group permissions already in the builder image to reduce
+# the size of the final image.
+# see https://github.com/stackabletech/docker-images/issues/961
+chmod -R g=u "${HOME}"
+chmod +x /stackable/opensearch/opensearch-docker-entrypoint.sh
+EOF
+
+
+# The OpenSearch Performance Analyzer needs a JDK, not just a JRE.
+# With a JRE, the following exception is thrown:
+# java.lang.ClassNotFoundException: com.sun.tools.attach.VirtualMachine
+FROM stackable/image/jdk-base
+
+ARG PRODUCT
+ARG RELEASE
+ARG OPENSEARCH_SECURITY_PLUGIN
+ARG STACKABLE_USER_UID
+
+ARG NAME="OpenSearch"
+ARG DESCRIPTION="This image is deployed by a custom Helm Chart"
+ARG HOME=/stackable
+ARG OPENSEARCH_HOME=${HOME}/opensearch
+
+LABEL name="${NAME}"
+LABEL version="${PRODUCT}"
+LABEL release="${RELEASE}"
+LABEL summary="The Stackable image for OpenSearch"
+LABEL description="${DESCRIPTION}"
+
+# https://github.com/opencontainers/image-spec/blob/036563a4a268d7c08b51a08f05a02a0fe74c7268/annotations.md#annotations
+LABEL org.opencontainers.image.version="${PRODUCT}"
+LABEL org.opencontainers.image.revision="${RELEASE}"
+LABEL org.opencontainers.image.title="${NAME}"
+LABEL org.opencontainers.image.description="${DESCRIPTION}"
+
+COPY \
+    --chown=${STACKABLE_USER_UID}:0 \
+    --from=opensearch-builder \
+    /stackable/opensearch \
+    /stackable/opensearch-${PRODUCT}-stackable${RELEASE}
+COPY \
+    --chown=${STACKABLE_USER_UID}:0 \
+    --from=opensearch-builder \
+    /stackable/opensearch-security \
+    /stackable/opensearch-${PRODUCT}-stackable${RELEASE}/plugins/opensearch-security
+COPY \
+    --chown=${STACKABLE_USER_UID}:0 \
+    --from=opensearch-builder \
+    /stackable/opensearch-${PRODUCT}-stackable${RELEASE}-src.tar.gz \
+    /stackable
+COPY \
+    --chown=${STACKABLE_USER_UID}:0 \
+    --from=opensearch-security-plugin \
+    /stackable/opensearch-security-${OPENSEARCH_SECURITY_PLUGIN}-stackable${RELEASE}-src.tar.gz \
+    /stackable
+COPY \
+    --chown=${STACKABLE_USER_UID}:0 \
+    --from=opensearch-builder \
+    /stackable/src/opensearch/patchable-work/worktree/${PRODUCT}/build/reports/bom.json \
+    /stackable/opensearch-${PRODUCT}-stackable${RELEASE}.cdx.json
+COPY \
+    --chown=${STACKABLE_USER_UID}:0 \
+    --from=opensearch-security-plugin \
+    /stackable/src/opensearch/security-plugin/patchable-work/worktree/${OPENSEARCH_SECURITY_PLUGIN}/build/reports/bom.json \
+    /stackable/opensearch-security-${OPENSEARCH_SECURITY_PLUGIN}-stackable${RELEASE}.cdx.json
+RUN <<EOF
+microdnf update
+microdnf clean all
+rm -rf /var/cache/yum
+
+# All files and folders owned by root group to support running as arbitrary users.
+# This is best practice as all container users will belong to the root group (0).
+chown -R ${STACKABLE_USER_UID}:0 ${HOME}
```

Comment:
Why don't you use the `check-permissions-ownership.sh` script? This layer is okay for now but if somebody else add things with other uids/gids, it could grow.

Reply:
Recursively changing the ownership could blow the image layer and undermines the purpose of the `check-permissions-ownership.sh` script. I would selectively change the files and directories with wrong ownership and permissions and let the `check-permissions-ownership.sh` script ensure, that everything is set correctly.

```shell
chown -h ${STACKABLE_USER_UID}:0 ${OPENSEARCH_HOME}
chmod g=u /stackable/opensearch-${PRODUCT}-stackable${RELEASE}
chmod g=u /stackable/opensearch-${PRODUCT}-stackable${RELEASE}/plugins/opensearch-security
```